src/cmd/flash.js

```diff
@@ -17,153 +16,132 @@ const systemModuleIndexToString = {
 };
 
 class FlashCommand {
-	constructor(options) {
-		this.options = options;
-	}
-
-	flash() {
-		if (!this.options.params.device) {
+	flash(device, binary, files, { usb, serial, factory, force, target, yes }) {
+		if (!device && !binary) {
 			// if no device nor files are passed, show help
-			return when.reject();
+			// TODO: Replace by UsageError
```

Comment:
TODO

Reply:
This works, but it's a quirk of the implementation that I'd like to make explicit: if a command rejects without a reason (like `return Promise.reject()`) the command processor will show usage information. We should have a shared UsageError class that can be thrown to accomplish this.